hooks/useBalances.ts

```diff
@@ -63,9 +66,17 @@ interface BalanceState {
   tokens: TokenBalance[];
   isLoading: boolean;
   isRefreshing: boolean;
+  isStale: boolean;
   error: string | null;
+  lastFetchTime: number | null;
+  retryCount: number;
 }
 
+const CACHE_DURATION = 30000;
```

Comment:
@LiorAgnin manual cache, retry etc. logic can be replaced with Tanstack query, it has all these data-fetching features

Reply:
@MusabShakeel576 Fixed.